lua/telescope/actions/set.lua

```diff
@@ -50,71 +49,65 @@ set.edit = function(prompt_bufnr, command)
-    if entry.filename then
-      filename = entry.path or entry.filename
-
-      -- TODO: Check for off-by-one
-      row = entry.row or entry.lnum
-      col = entry.col
-    elseif not entry.bufnr then
-      -- TODO: Might want to remove this and force people
-      -- to put stuff into `filename`
-      local value = entry.value
-      if not value then
-        print("Could not do anything with blank line...")
-        return
-      end
-
-      if type(value) == "table" then
-        value = entry.display
-      end
-
-      local sections = vim.split(value, ":")
+  end
 
-      filename = sections[1]
-      row = tonumber(sections[2])
-      col = tonumber(sections[3])
+  local filename, row, col
+
+  if entry.filename then
+    filename = entry.path or entry.filename
+
+    -- TODO: Check for off-by-one
+    row = entry.row or entry.lnum
+    col = entry.col
+  elseif not entry.bufnr then
+    -- TODO: Might want to remove this and force people
+    -- to put stuff into `filename`
+    local value = entry.value
+    if not value then
+      print("Could not do anything with blank line...")
+      return
     end
 
-    local preview_win = state.get_status(prompt_bufnr).preview_win
-    if preview_win then
-      a.nvim_win_set_config(preview_win, {style = ''})
+    if type(value) == "table" then
+      value = entry.display
     end
 
-    local entry_bufnr = entry.bufnr
+    local sections = vim.split(value, ":")
 
-    require('telescope.actions').close(prompt_bufnr)
+    filename = sections[1]
+    row = tonumber(sections[2])
+    col = tonumber(sections[3])
+  end
 
-    if entry_bufnr then
-      if command == 'edit' then
-        vim.cmd(string.format(":buffer %d", entry_bufnr))
-      elseif command == 'new' then
-        vim.cmd(string.format(":sbuffer %d", entry_bufnr))
-      elseif command == 'vnew' then
-        vim.cmd(string.format(":vert sbuffer %d", entry_bufnr))
-      elseif command == 'tabedit' then
-        vim.cmd(string.format(":tab sb %d", entry_bufnr))
-      end
-    else
-      filename = path.normalize(vim.fn.fnameescape(filename), vim.fn.getcwd())
-
-      local bufnr = vim.api.nvim_get_current_buf()
-      if filename ~= vim.api.nvim_buf_get_name(bufnr) then
-        vim.cmd(string.format(":%s %s", command, filename))
-        bufnr = vim.api.nvim_get_current_buf()
-        a.nvim_buf_set_option(bufnr, "buflisted", true)
-      end
+  local entry_bufnr = entry.bufnr
+
+  require('telescope.actions').close(prompt_bufnr)
+
+  if entry_bufnr then
+    if command == 'edit' then
+      vim.cmd(string.format(":buffer %d", entry_bufnr))
+    elseif command == 'new' then
+      vim.cmd(string.format(":sbuffer %d", entry_bufnr))
+    elseif command == 'vnew' then
+      vim.cmd(string.format(":vert sbuffer %d", entry_bufnr))
+    elseif command == 'tabedit' then
+      vim.cmd(string.format(":tab sb %d", entry_bufnr))
+    end
+  else
+    filename = path.normalize(vim.fn.fnameescape(filename), vim.fn.getcwd())
```

Comment:
We kinda wanna want speed now so we should use `vim.loop.cwd` :D no more viml :rofl: